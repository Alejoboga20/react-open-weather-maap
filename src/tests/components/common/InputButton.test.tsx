import { shallow } from 'enzyme';
import {
  InputButton,
  InputButtonProps,
} from '../../../components/common/InputButton';

const inputButtonProps: InputButtonProps = {
  label: 'test label',
  value: 'test balue',
  location: { latitude: 10, longitude: 10 },
  onClick: jest.fn(),
  disabled: false,
};

describe('InputButton Tests', () => {
  const wrapper = shallow(<InputButton {...inputButtonProps} />);

  test('should Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have correct label', () => {
    const title = wrapper.find('button').text();

    expect(title).toBe(inputButtonProps.label);
  });

  test('should call onClick function', () => {
    wrapper.find('button').simulate('click');

    expect(inputButtonProps.onClick).toHaveBeenCalled();
  });
});
