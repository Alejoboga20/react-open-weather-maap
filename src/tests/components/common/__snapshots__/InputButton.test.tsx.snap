// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButton Tests should Match Snapshot 1`] = `
<button
  className="inputButton"
  disabled={false}
  onClick={[MockFunction]}
  value="test balue"
>
  test label
</button>
`;
